lib/logfmt/encoder.ex

```diff
@@ -14,40 +51,38 @@ defmodule Logfmt.Encoder do
   @doc """
   See [`Logfmt.encode`](/logfmt/Logfmt.html#encode/1).
   """
-  @spec encode(Dict.t) :: String.t
-  def encode(list) do
-    Enum.reduce list, "", &encode_pair/2
-  end
-
-  @spec encode_pair({Dict.key, String.t | boolean | number}, String.t) :: String.t
-  defp encode_pair({key, value}, "") when is_binary(value) do
-    if String.contains?(value, "=") do
-      ~s(#{key}="#{encode_value(value)}")
-    else
-      ~s(#{key}=#{encode_value(value)})
+  @spec encode(Dict.t, options :: Keyword.t) :: String.t
+  def encode(list, options \\ []) do
+    result = Enum.reduce(list, [], &encode_pair/2)
```

Comment:
Technically the reduce may be more efficient, but I think this may be a little more readable:

```
result =
  list
  |> Enum.map(&encode_pair/1)
  |> Enum.intersperse(" ")
```

Which also makes the `encode_pair/1` function not have to be aware of the accumulator, which is nice.

Reply:
And done, that was pretty neat, I've never used intersperse before